File: packages/shared-types/src/health.ts
/** Mirrors `app/schemas/health.py::HealthResponse`. */
export interface HealthResponse {
  status: "ok" | "degraded";
  service: string;
  version: string;
  environment: string;
  database: "ok" | "unavailable";
  storage: "ok" | "unavailable";
  /** Configured engine/provider names only (never keys or endpoints). */
  providers: Record<string, string>;
  uptime_seconds: number;
  /** Binary engines the pipeline shells out to (c2patool): status and version, never paths. */
  engines: Record<
    string,
    {
      status: "ok" | "unavailable" | "not_configured";
      version: string | null;
      /** Trust-list mode and version the provenance engine evaluates against (c2patool). */
      trust?: { mode: string; list_version: string | null };
    }
  >;
}

/** Mirrors `app/schemas/health.py::LivenessResponse`. */
export interface LivenessResponse {
  status: "ok";
}

File: services/api/app/schemas/health.py
from typing import Any, Literal

from pydantic import BaseModel


class LivenessResponse(BaseModel):
    status: Literal["ok"]


class HealthResponse(BaseModel):
    status: Literal["ok", "degraded"]
    service: str
    version: str
    environment: str
    database: Literal["ok", "unavailable"]
    storage: Literal["ok", "unavailable"]
    # Configured engine/provider names only (never keys or endpoints).
    providers: dict[str, str]
    uptime_seconds: float
    # Binary engines the pipeline shells out to: name -> {status, version}; never paths.
    engines: dict[str, dict[str, Any]] = {}
